mathesar/rpc/records.py

```diff
@@ -152,12 +140,13 @@ class RecordList(TypedDict):
         count: The total number of records in the table.
         results: An array of record objects.
         grouping: Information for displaying grouped records.
-        preview_data: Information for previewing foreign key values.
+        preview_data: Information for previewing foreign key values,
+            provides a map of foreign key to a text summary.
     """
     count: int
     results: list[dict]
     grouping: GroupingResponse
-    preview_data: dict[str, list[PreviewEntry]]
+    preview_data: dict[str, dict[Any, str]]
```

Comment:
The key will always be a string due to the nature of `jsonb_object_agg`. So, this should be:

```suggestion
    preview_data: dict[str, dict[str, str]]
```

@seancolsen To double-check, are you sure that mistyping won't cause a problem?